server/sonar-server/src/test/java/org/sonar/server/qualityprofile/ws/QProfileChangeParentActionMediumTest.java

```diff
@@ -0,0 +1,197 @@
+import org.sonar.server.db.DbClient;
+import org.sonar.server.qualityprofile.QProfileName;
+import org.sonar.server.qualityprofile.QProfileTesting;
+import org.sonar.server.tester.ServerTester;
+import org.sonar.server.user.MockUserSession;
+import org.sonar.server.ws.WsTester;
+
+import java.util.List;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class QProfileChangeParentActionMediumTest {
+
+  // TODO Replace with DbTester + EsTester once DaoV2 is removed
+  @ClassRule
+  public static ServerTester tester = new ServerTester();
+
+  QProfilesWs ws;
+  DbClient db;
+  DbSession session;
+  WsTester wsTester;
+
+  @Before
+  public void setUp() throws Exception {
+    tester.clearDbAndIndexes();
+    db = tester.get(DbClient.class);
+    ws = tester.get(QProfilesWs.class);
+    wsTester = tester.get(WsTester.class);
+    session = db.openSession(false);
+    MockUserSession.set().setLogin("gandalf").setGlobalPermissions(GlobalPermissions.QUALITY_PROFILE_ADMIN);
+  }
+
+  @After
+  public void after() {
+    session.close();
+  }
+
+  @Test
+  public void change_parent_with_keys() throws Exception {
+    QualityProfileDto parent1 = createProfile("xoo", "Parent 1");
+    QualityProfileDto parent2 = createProfile("xoo", "Parent 2");
+    QualityProfileDto child = createProfile("xoo", "Child");
+
+    RuleDto rule1 = createRule("xoo", "rule1");
+    RuleDto rule2 = createRule("xoo", "rule2");
+    createActiveRule(rule1, parent1);
+    createActiveRule(rule2, parent2);
+    session.commit();
+
+    assertThat(db.activeRuleDao().findByProfileKey(session, child.getKey())).isEmpty();
+
+    // 1. Set parent 1
+    wsTester.newGetRequest(QProfilesWs.API_ENDPOINT, "change_parent")
+      .setParam(QProfileIdentificationParamUtils.PARAM_PROFILE_KEY, child.getKey().toString())
+      .setParam("parentKey", parent1.getKey().toString())
+      .execute();
+    session.clearCache();
+
+    // 1. Assert ActiveRule in DAO
+    List<ActiveRuleDto> activeRules1 = db.activeRuleDao().findByProfileKey(session, child.getKey());
+    assertThat(activeRules1).hasSize(1);
+    assertThat(activeRules1.get(0).getKey().ruleKey().rule()).isEqualTo("rule1");
+
+    // 2. Set parent 2
+    wsTester.newGetRequest(QProfilesWs.API_ENDPOINT, "change_parent")
+      .setParam(QProfileIdentificationParamUtils.PARAM_PROFILE_KEY, child.getKey().toString())
+      .setParam("parentKey", parent2.getKey().toString())
+      .execute();
+    session.clearCache();
+
+    // 2. Assert ActiveRule in DAO
+    List<ActiveRuleDto> activeRules2 = db.activeRuleDao().findByProfileKey(session, child.getKey());
+    assertThat(activeRules2).hasSize(1);
+    assertThat(activeRules2.get(0).getKey().ruleKey().rule()).isEqualTo("rule2");
+
+    // 3. Remove parent
+    wsTester.newGetRequest(QProfilesWs.API_ENDPOINT, "change_parent")
+      .setParam(QProfileIdentificationParamUtils.PARAM_PROFILE_KEY, child.getKey().toString())
+      .execute();
+    session.clearCache();
```

Comment:
A removal test is missing when using empty parent : setParam("parentKey", ""), and it's currently failing with "Quality profile not found: ", but it should not, it should remove the parent.